src/Symfony/Component/PropertyAccess/PropertyAccessor.php

```diff
@@ -428,17 +428,23 @@ private function readProperty(array $zval, string $property, bool $ignoreInvalid
                 // handle uninitialized properties in PHP >= 7.4
                 if (\PHP_VERSION_ID >= 70400 && preg_match('/^Typed property ([\w\\\]+)::\$(\w+) must not be accessed before initialization$/', $e->getMessage(), $matches)) {
                     $r = new \ReflectionProperty($matches[1], $matches[2]);
-                    $type= ($type = $r->getType()) instanceof \ReflectionNamedType ? $type->getName() : (string) $type;
+                    $type = ($type = $r->getType()) instanceof \ReflectionNamedType ? $type->getName() : (string) $type;
 
                     throw new UninitializedPropertyException(sprintf('The property "%s::$%s" is not readable because it is typed "%s". You should initialize it or declare a default value instead.', $r->getDeclaringClass()->getName(), $r->getName(), $type), 0, $e);
                 }
 
                 throw $e;
             }
-        } elseif ($object instanceof \stdClass && property_exists($object, $property)) {
-            $result[self::VALUE] = $object->$property;
-            if (isset($zval[self::REF])) {
-                $result[self::REF] = &$object->$property;
+        } elseif (property_exists($object, $property)) {
+            try {
+                $result[self::VALUE] = $object->$property;
+                if (isset($zval[self::REF])) {
+                    $result[self::REF] = &$object->$property;
+                }
+            } catch (\Error $e) {
+                if (!$ignoreInvalidProperty) {
```

Comment:
maybe it need also a check on the error message (like done upper) wdyt @nicolas-grekas ?

Reply:
sure

Reply:
done